pulp_container/tests/functional/api/test_pull_content.py

```diff
@@ -309,94 +206,44 @@ def setUpClass(cls):
-        cls.remotes_api = RemotesContainerApi(client_api)
-        cls.distributions_api = DistributionsContainerApi(client_api)
-
-        cls.teardown_cleanups = []
-
-        delete_orphans()
-
-        with contextlib.ExitStack() as stack:
-            # ensure tearDownClass runs if an error occurs here
-            stack.callback(cls.tearDownClass)
-
-            # Step 1
-            _repo = cls.repositories_api.create(ContainerContainerRepository(**gen_repo()))
-            cls.teardown_cleanups.append((cls.repositories_api.delete, _repo.pulp_href))
-
-            # Step 2
-            cls.remote = cls.remotes_api.create(gen_container_remote(policy="on_demand"))
-            cls.teardown_cleanups.append((cls.remotes_api.delete, cls.remote.pulp_href))
-
-            # Step 3
-            sync_data = ContainerRepositorySyncURL(remote=cls.remote.pulp_href)
-            sync_response = cls.repositories_api.sync(_repo.pulp_href, sync_data)
-            monitor_task(sync_response.task)
-
-            cls.repo = cls.repositories_api.read(_repo.pulp_href)
-            cls.artifacts_api = ArtifactsApi(core_client)
-            cls.artifact_count = cls.artifacts_api.list().count
-
-            # Step 4.
-            distribution_response = cls.distributions_api.create(
-                ContainerContainerDistribution(**gen_distribution(repository=cls.repo.pulp_href))
-            )
-            created_resources = monitor_task(distribution_response.task).created_resources
-
-            distribution = cls.distributions_api.read(created_resources[0])
-            cls.distribution_with_repo = cls.distributions_api.read(distribution.pulp_href)
-            cls.teardown_cleanups.append(
-                (cls.distributions_api.delete, cls.distribution_with_repo.pulp_href)
-            )
-
-            # Step 5.
-            distribution_response = cls.distributions_api.create(
-                ContainerContainerDistribution(
-                    **gen_distribution(repository_version=cls.repo.latest_version_href)
-                )
-            )
-            created_resources = monitor_task(distribution_response.task).created_resources
-            distribution = cls.distributions_api.read(created_resources[0])
-            cls.distribution_with_repo_version = cls.distributions_api.read(distribution.pulp_href)
-            cls.teardown_cleanups.append(
-                (cls.distributions_api.delete, cls.distribution_with_repo_version.pulp_href)
-            )
-
-            # remove callback if everything goes well
-            stack.pop_all()
-
-    @classmethod
-    def tearDownClass(cls):
-        """Clean class-wide variable."""
-        for cleanup_function, args in reversed(cls.teardown_cleanups):
-            cleanup_function(args)
-
-    def test_api_returns_same_checksum(self):
+        monitor_task(
+            pulpcore_bindings.OrphansCleanupApi.cleanup({"orphan_protection_time": 0}).task
+        )
```

Comment:
We do have the orphan_cleanup_pre fixture for this I believe.
And it even prevents you from adding the parallel mark.

Reply:
It's called `delete_orphans_pre`, yeah?

Reply:
I already tried, it is not allowed because of scope mistmatch.